Bottom: b6b89a7c837b7330f8cd340c75d1746ad265f3d4
Top:    5188c490b5cf38c9cc37c248447020c853cf6a75
Author: Ben Greear <greearb@candelatech.com>
Date:   2013-05-09 11:56:17 -0700

net: Ensure tx watchdog failures always print.

These are too important to not have some log message
generated.  But, keep the logic that only prints
a single stack dump.

Signed-off-by: Ben Greear <greearb@candelatech.com>


---

diff --git a/net/sched/sch_generic.c b/net/sched/sch_generic.c
index 53e6dcd..d270536 100644
--- a/net/sched/sch_generic.c
+++ b/net/sched/sch_generic.c
@@ -266,8 +266,14 @@ static void dev_watchdog(unsigned long arg)
 			}
 
 			if (some_queue_timedout) {
-				WARN_ONCE(1, KERN_INFO "NETDEV WATCHDOG: %s (%s): transmit queue %u timed out\n",
-				       dev->name, netdev_drivername(dev), i);
+				printk(KERN_INFO "NETDEV WATCHDOG: %s (%s):"
+				       " transmit queue %u timed out,"
+				       " trans_start: %lu, wd-timeout: %i"
+				       " jiffies: %lu tx-queues: %i\n",
+				       dev->name, netdev_drivername(dev), i,
+				       trans_start, dev->watchdog_timeo,
+				       jiffies, dev->num_tx_queues);
+				WARN_ON_ONCE(1);
 				dev->netdev_ops->ndo_tx_timeout(dev);
 			}
 			if (!mod_timer(&dev->watchdog_timer,
